src/dotnet/Chat.Service/ChatAuthors.cs

```diff
@@ -96,19 +97,33 @@ public virtual async Task<ImmutableArray<Symbol>> ListChatIds(Session session, C
 
     // [ComputeMethod]
     public virtual async Task<Author?> GetAuthor(
-        string chatId, string authorId, bool inherit,
+        Session session,
+        string chatId,
+        string authorId,
+        bool inherit,
         CancellationToken cancellationToken)
     {
+        // Check that user has access to chat
+        var chat = await Chats.Get(session, chatId, cancellationToken).ConfigureAwait(false);
+        if (chat == null)
```

Comment:
I think we have the same behavior for GetAuthor and GetAuthorPresence if we can not read the chat.
Extract method that checks if we can read the chat.

Reply:
done